src/transformers/models/zoedepth/image_processing_zoedepth.py

```diff
@@ -442,3 +449,129 @@ def preprocess(
 
         data = {"pixel_values": images}
         return BatchFeature(data=data, tensor_type=return_tensors)
+
+    def post_process_depth_estimation(
+        self,
+        outputs,
+        source_sizes: Union[TensorType, List[Tuple[int, int]]],
+        target_sizes: Optional[Union[TensorType, List[Tuple[int, int]], None]] = None,
```

Comment:
From the code below - I'd have expected source_sizes to default to None and target_sizes to be required

Reply:
`source_sizes` is required because it is used to calculate the padding to be removed from around the depth map. `target_sizes` is only used to an additional resizing of the result (and also to match the rest of the `post_process_...` functions.

Reply:
If `source_sizes` is required, then we should remove the `None` handling logic on L537

Reply:
It is required only if the user has not explicitly told the preprocessor *not* to add padding. The `None` option for the `source_sizes`, tied with the `do_remove_padding` input, can be used in conjunction if users pass the `do_pad=False` flag in the image processor so that during post-processing we do not try to remove non-existent padding.

Reply:
In this case - this should be reflected in the docstring i.e. `source_sizes (`TensorType` or `List[Tuple[int, int]]`, *optional*):`

and then in the description say when it is required